region/client.go

```diff
@@ -365,67 +364,62 @@ func (c *client) processRPCs() {
 		m = newMulti(c.rpcQueueSize)
 	}
 
-	for {
-		// first loop is to accomodate request heavy workload
-		// it will batch as long as conccurent writers are sending
-		// new rpcs or until multi is filled up
-		for {
-			select {
-			case <-c.done:
-				return
-			case rpcs := <-c.rpcs:
-				// have things queued up, batch them
-				if !m.add(rpcs) {
-					// can still put more rpcs into batch
-					continue
-				}
-			default:
-				// no more rpcs queued up
-			}
-			break
+	var t *time.Timer
+	if c.flushInterval > 0 {
+		// Initialize timer and then stop it. It will be Reset later
+		t = time.NewTimer(0)
+		if !t.Stop() {
+			<-t.C
 		}
+	}
 
-		if l := m.len(); l == 0 {
-			// wait for the next batch
-			select {
-			case <-c.done:
-				return
-			case rpcs := <-c.rpcs:
-				m.add(rpcs)
+batchloop:
+	for {
+		// wait for first rpc
+		select {
+		case <-c.done:
+			return
+		case rpcs := <-c.rpcs:
+			if m.add(rpcs) {
+				flush("queue full")
+				continue
 			}
-			continue
-		} else if l >= c.rpcQueueSize || c.flushInterval == 0 {
-			// batch is full, flush
-			flush("queue full")
-			continue
 		}
-
-		// second loop is to accomodate less frequent callers
-		// that would like to maximize their batches at the expense
-		// of waiting for flushInteval
-		timer := time.NewTimer(c.flushInterval)
-		reason := ""
-		for {
-			select {
-			case <-c.done:
-				return
-			case <-timer.C:
-				reason = "timeout"
-				// time to flush
-			case rpcs := <-c.rpcs:
-				if !m.add(rpcs) {
-					// can still put more rpcs into batch
-					continue
+		if c.flushInterval > 0 {
```

Comment:
I don't think that works because of the `default` case in the `select`. It's required when `flushInterval == 0`, but would cause a busy loop if when `flushInterval > 0`.